crypto/crypto.go

```diff
@@ -90,28 +98,38 @@ func HashData(kh KeccakState, data []byte) (h common.Hash) {
 
 // Keccak256 calculates and returns the Keccak256 hash of the input data.
 func Keccak256(data ...[]byte) []byte {
-	b := make([]byte, 32)
-	d := hasherPool.Get().(KeccakState)
+	b := Keccak256Hash(data...)
+	return b[:]
+}
+
+// Keccak256RLP calculates and returns the Keccak256 hash of the input interface.
+func Keccak256RLP(x interface{}) []byte {
+	b := Keccak256RLPHash(x)
+	return b[:]
+}
+
+// Keccak256Hash calculates and returns the Keccak256 hash of the input data,
+// converting it to an internal Hash data structure.
+func Keccak256Hash(data ...[]byte) common.Hash {
+	d := hasherPool.Get().(*keccakState)
 	d.Reset()
 	for _, b := range data {
 		d.Write(b)
 	}
-	d.Read(b)
+	d.Read(d.hashBuf[:])
 	hasherPool.Put(d)
-	return b
+	return d.hashBuf
 }
 
-// Keccak256Hash calculates and returns the Keccak256 hash of the input data,
+// Keccak256RLPHash calculates and returns the Keccak256 hash of the input interface,
 // converting it to an internal Hash data structure.
-func Keccak256Hash(data ...[]byte) (h common.Hash) {
-	d := hasherPool.Get().(KeccakState)
+func Keccak256RLPHash(x interface{}) common.Hash {
+	d := hasherPool.Get().(*keccakState)
 	d.Reset()
-	for _, b := range data {
-		d.Write(b)
-	}
-	d.Read(h[:])
+	rlp.Encode(d, x)
+	d.Read(d.hashBuf[:])
```

Comment:
Don't we override the hashbuffer with this, making it unusuable for the next caller?

Reply:
Sorry, I don't understand your point. Do you mean the `hashBuf` can't be used at the next call?